### ISSUE TITLE
What's the reason for no dissoc_cursor in WriteTransaction?

### ISSUE THREAD

**Issue by yrashk** [AltSysrq/lmdb-zero#6]
I am not sure I am getting the full picture yet and I noticed that WriteTransaction doesn't have cursor disassociation. I presume it's because write transaction might modify something?

**Comment by AltSysrq** [comment]
LMDB itself only allows reusing cursors across read-only transactions (see [mdb_cursor_renew](http://www.lmdb.tech/doc/group__mdb.html#gac8b57befb68793070c85ea813df481af)). This is because cursors from write transactions are destroyed implicitly when the transaction is destroyed, which I believe is because aborting or committing a transaction may invalidate memory that the cursors were using for bookkeeping (not entirely sure on the specifics of this, though).

**Comment by yrashk** [comment]
I am wondering because (as you might remember) I am passing lmdb-related data structures around and while I was able to pass StaleCursors easily, I am tempted to see if I can use `cursor::to_stale` to WriteTransaction's cursors, given the fact that I know that I *will* remove all cursors upon write transaction termination.

**Comment by AltSysrq** [comment]
Oh, so you're trying to work around the lifetimes. I have a branch I've been working on for some time that will make that more flexible which I hope to have in soonish (maybe this weekend if nothing comes up).

In the mean time, if you're fine taking things into your own hands and introducing unsafety, you can just cast the lifetime away:

```rust
unsafe fn cast_away(cursor: lmdb::Cursor) -> lmdb::Cursor<'static,'static> {
  ::std::mem::transmute(cursor)
}
```

Just keep in mind that if you do this and fail to ensure the cursor is dropped before the transaction you'll get undefined behaviour.

**Comment by yrashk** [comment]
Gotcha! Thanks. I will try this for WriteTransaction soon. And I am definitely looking forward to your branch. Let me know if you can you use any help or if you want to test it on me first before merging :) Thanks!

**Comment by yrashk** [comment]
P.S. The `cast_away` solution worked wonderfully. Thanks for pointing it out, I am still getting up to speed with Rust, especially the evil tricks.

**Comment by AltSysrq** [comment]
I believe the root issue here was solved with the more flexible 0.4.0 API, closing. Feel free to reopen if this is not the case.
